specs/execution-api.md

```diff
@@ -0,0 +1,95 @@
+# Execution API Specification
+
+## Overview
+
+The Execution API is the interface `Conductor` uses to drive deterministic execution of sequenced blocks in dependent chains. Inspired by other APIs, such as the Engine API and ABCI Consensus API, it is a chain agnostic mechanism intended to be very simple to implement. It is a gRPC API which any state machine can implement and use conductor with to drive their execution to integrate with the Astria Sequencer.
+
+> Note: this documentation is for the v1alpha2 API which is currently being implemented, and some of the documentation is on how it should be implemented, not how it is currently.
+
+## Basic Design Principles
+
+The Execution API is a resource based API with two resources: `Block` and `CommitmentState`. The API is designed to follow basic principles outlined by aip.dev as best practices for resource based APIs. gRPC has been chosen for the API due to the wide availability of language implementations which make it easy to generate client libraries, as well as stubs for server implementations.  
+
+## Conductor Usage
+
+### Startup
+
+Upon startup, conductor needs to know the state of commitments in the state machine and calls `GetCommitmentState`. If started on a fresh rollup these will all be the same block. If running against a state machine with previous block data, Conductor must also track the block hash of any blocks between commitments, it will call `BatchGetBlock` to get block information between commitments.
+
+### Execution & Commitments
+
+From the perspective of the sequencer:
+- `HEAD` commitments are made every time the sequencer creates a new block at height N. Only the `HEAD` blocks can be reorged and they can be updated until the N+1 block has been created. Once the N+1 block is received, the block at height N is set to `SOFT`
+- `SOFT` commitment means that sequencer consensus has full agreement on the block.
+- `FIRM` commitment indicates that the block has been written, and has been propogated across the DA network.
```

Comment:
```suggestion
- `FIRM` commitment indicates that the block has been written and has been propagated across the DA network.
```
The comma suggestion is just a nit. You shouldn't need a comma when combining an independent and dependent clause.